### DISCUSSION TITLE
In RT simulation, how to confirm the valid receiver position? (For example, not inside a building)

### DISCUSSION THREAD

**Discussion by wang-chenlong** [NVlabs/sionna#689] [General]
I’m working on large-scale RT simulations of mixed urban and suburban environments, analyzing multipath propagation at each receiver point. However, the `coverage_map` doesn’t account for terrain information and doesn’t provide multipath calculations. As a result, I plan to use `compute_paths` to calculate all the sample points in the scene. The challenge is that there doesn’t seem to be a straightforward way to ensure receiver positions are valid, i.e., located outside of buildings, and neither Sionna nor Mitsuba offers a direct method for this.

Any help is appreciated! Thanks in advance!

**Comment by hieutrungle** [comment]
I think of a workaround: 

1. Enable the diffraction argument in the compute_paths method.
2. Analyze the resulting paths:
 - Points that receive reflection paths, even small ones, can be considered visible.
 - Points that do not receive any paths are likely hidden, inside buildings, or extremely far away from transmitters.

This uses the fact that most visible points will have some form of reflection or diffraction paths, while completely occluded points will not. This method may not be 100% accurate in all scenarios, especially for complex geometries or edge cases

**Comment by SebastianCa** [comment]
Hi @wang-chenlong,
a possible solution is to [sample the receiver positions](https://nvlabs.github.io/sionna/api/rt.html#sionna.rt.CoverageMap.sample_positions) directly from the coverage map as shown in the [tutorial notebook](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Introduction.html#Site-specifc-Link-Level-Simulations). If the receiver positions are given, you can use the coverage map to verify that the received signal strength is above a certain threshold.

I’ll transfer this issue to Discussions, as it is not a technical issue with Sionna.

**Comment by wang-chenlong** [comment]
@SebastianCa @hieutrungle ,
Thank you very much for your help.
I found that in `paths,` if a point is inside a building, the output from `compute_path` is empty. However, if the transmitter and receiver are far apart with no multipath propagation, the result is also empty.
Ultimately, my solution was to map the Blender coordinates to the 4326 coordinate system, then use `osmnx` to extract OSM building data, and use the `shapely` library to check if a point is inside a building.
